pkg/apis/certificates/validation/validation.go

```diff
@@ -205,6 +206,10 @@ func validateCertificateSigningRequest(csr *certificates.CertificateSigningReque
 	} else {
 		allErrs = append(allErrs, ValidateCertificateSigningRequestSignerName(specPath.Child("signerName"), csr.Spec.SignerName)...)
 	}
+	const min = 5 * time.Minute
+	if csr.Spec.DurationHint != nil && csr.Spec.DurationHint.Duration < min {
+		allErrs = append(allErrs, field.Invalid(specPath.Child("durationHint"), csr.Spec.DurationHint.Duration, "may not specify a duration less than 5 minutes"))
```

Comment:
What was the rationale for a minimum of 5 minutes?

Reply:
For reference, the projected token lifetime minimum is 10 minutes, and kubelet renewals start at 80% of that, so 8 minutes. Given CSR approval/signing is async, I'd expect to leave at least as much time for a CSR renewal request to be processed... starting renewal 3 minutes after issuance seems aggressive.